fitbenchmarking/fitbenchmarking
# Split parsing and running so that names can be made unique

#### Description of Work
Run parsing as a seperate step to allow changing the name of the problems to ensure they're unique.

This will mean that all problems are loaded into memory before any are run which I was concerned about until I realised that problems are kept in memory after running anyway. 

Fixes #408 


#### Testing Instructions

1. Rename 2 problems in a problem directory to have the same name.
2. Run fitbenchmarking
3. Check that 2 results are generated

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?
